src/gate/library/parser_lib.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "kitty/kitty.hpp"
+#include "nlohmann/json.hpp"
+
+using json = nlohmann::json;
+
+namespace eda::gate::library {
+
+class LibertyParser {
+
+private:
+  json jFile;
+  kitty::dynamic_truth_table *cell;
+
+  void writeCell(std::vector<std::string> inputPins, 
+      std::string cellName, std::string function);
+
+public:
+  std::vector <kitty::dynamic_truth_table> cells;
```

Comment:
Public field is suspicious: is it a part of the interface? Then what is the protocol and how it is enforced?